client/src/components/DataFiles/DataFilesModals/DataFilesUploadModal.js

```diff
@@ -80,18 +84,28 @@ const DataFilesUploadModal = ({ className, layout }) => {
 
   const selectFiles = acceptedFiles => {
     const newFiles = [];
-    acceptedFiles.forEach(file => {
+    const newAcceptedFiles = acceptedFiles.filter(
+      af =>
+        uploadedFiles.filter(
+          uf => uf.data.path === af.path && uf.data.size === af.size
+        ).length === 0
+    );
+    newAcceptedFiles.forEach(file => {
       newFiles.push({ data: file, id: uuidv4() });
     });
```

Comment:
No change needed, but we can also chain array methods
```suggestion
    const newAcceptedFiles = acceptedFiles
      .filter(
        ({ path, size }) =>
          !uploadedFiles.some(
            uf => uf.data.path === path && uf.data.size === size
          )
      )
      .map(file => ({ data: file, id: uuidv4() }));
```